### PR TITLE
[ENH] Migrating ResNet Regressor from sktime_dl

### PR DESCRIPTION
**Pull request by nilesh05apr**
#### Reference Issues/PRs
For : [ENH] Migrating Estimators from sktime-dl #3351

#### What does this implement/fix? Explain your changes.
Migrated ResNet  Regressor from sktime_dl to sktime

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### What should a reviewer concentrate their feedback on?
Tests

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.
- [ ] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by nilesh05apr** [comment]
@achieveordie can you please review the changes, i added resnet regressor.

**Comment by achieveordie** [comment]
Hey, @nilesh05apr thanks for adding this as well! I really appreciate it.

However, it seems like you started this PR directly from the last one. Doing this might result in conflicts as you may be able to see it in the current [webpage](https://github.com/sktime/sktime/pull/3928/conflicts). If the current PR has conflicts from the main, GitHub Actions won't run the tests.

The ideal workflow may be to have two remote URLs, origin and upstream (the difference between them can be understood from [this](https://stackoverflow.com/a/9257901/13170092)):
1. Before working on a PR, make sure that your origin's main is updated to match the upstream's main (to ensure your fork has all newly made changes by other contributors).
2. Create a new branch to branch off from your repo's main and make changes in this branch.
3. Finally, make a PR to push commits onto `sktime:main`.
4. Optionally, delete the old branch after PR has been merged.

So while we can resolve the conflict here, I urge you to consider following the above workflow. This will improve your git skills and makes your life easier by avoiding unnecessary merge conflicts.

**Comment by nilesh05apr** [comment]
@achieveordie thank you for your help. I was unaware of this fact. I will follow the workflow as you mentioned. Also can you please resolve this merge conflict?

**Comment by achieveordie** [comment]
When you made the commits locally, did `pre-commit` not run?

**Comment by nilesh05apr** [comment]
no, there was no activity.

**Comment by achieveordie** [comment]
I recommend opening an issue if that's the case. Add all the info that you think might help in debugging it.

**Comment by fkiraly** [comment]
@nilesh05apr, I quickly looked into the failures, they seem genuine.

The failing test tells you that you are mutating the parameter `callbacks` in `_fit`.

Generally, as per the `sklearn` interface, arguments of `__init__` should be written to attributes of the same name, and then never changed by anymethod.

**Review by achieveordie — CHANGES_REQUESTED** [review]
Generally speaking, the best way to port estimators is by looking at how similar estimators have been ported.

If you think that there's a better way to implement any functionality, write it in your code, see if all the tests pass and then propose your new way as an alternative with all the pros (and cons) that come with it.

**Comment by achieveordie** [comment]
Nothing major, only minor changes are required to pass the tests and improve overall code clarity. Good work!

**Comment by fkiraly** [comment]
please comment in open discussions how the requests have been addressed

**Comment by nilesh05apr** [comment]
@achieveordie why test is not completed any help?

**Comment by achieveordie** [comment]
Probably because a new workflow `test-mlflow` was added in main but your branch doesn't have the files to run it. I'd recommend merging main onto this branch so all the newly added files are present in your fork.

**Comment by fkiraly** [comment]
just updated from `main` and restarted the tests.

**Comment by nilesh05apr** [comment]
@achieveordie how to fix this memory error, i have never seen this before.
can you please help?

**Comment by achieveordie** [comment]
Looking at your fork, there are two possibilities - 
1. It is being caused by `LSTMFCNClassifier`, see #4033 (your fork contains this estimator and test for it has only been minimally skipped)
2. It is caused by `ResNetRegressor` - which hints that there's a common root cause.

In any case, I'm solving this bug in #4036 and to solve it, there's another bug that needs to be addressed first. I'll add an issue and PR regarding it soon.

In the meantime, I'd encourage you to look around these errors and see if you find something interesting which can be useful to debug this issue.

File: sktime/regression/deep_learning/resnet.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        self.batch_size = batch_size
+        self.random_state = random_state
+        self.verbose = verbose
+        self.optimizer = optimizer
+        self.loss = loss
+        self.metrics = metrics
+        self.activation = activation
+        self.use_bias = use_bias
+        self.callbacks = callbacks
+        self._network = ResNetNetwork()
```
Comment (achieveordie) [inline sktime/regression/deep_learning/resnet.py:79]: Please update this to reflect the changes as suggested by #4075.

**Review by achieveordie — CHANGES_REQUESTED** [review]

File: sktime/regression/deep_learning/resnet.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        X   : np.ndarray of shape = (n_instances(n), n_dimensions(d), series_length(m))
+            Input training samples
+        y   : np.ndarray of shape n
+            Input training responses
+
+        Returns
+        -------
+        self: object
+        """
+        if self.callbacks is None:
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply sktime/regression/deep_learning/resnet.py:141]: well, #4095 is not yet the "right" fix due to the "not mutating parameters" contract violation

**Pull request closed without merging**